src/walk.rs

```diff
@@ -362,6 +362,16 @@ fn spawn_senders(
                 return ignore::WalkState::Continue;
             }
 
+            let mut ret_state = ignore::WalkState::Continue;
+            // when pruning, don't descend into matching dirs
+            // note that prune requires pattern
+            if config.prune
+                && pattern.as_str().len() != 0
+                && entry.file_type().map_or(false, |e| e.is_dir())
+            {
+                ret_state = ignore::WalkState::Skip;
+            }
```

Comment:
Please make `ret_state` non-mutable.  Instead, use something like:
```rs
let walk_action = if … {
  ignore::WalkState::Skip
} else {
  ignore::WalkState::Continue
};
```
Also, let's rename `ret_state` into `walk_action` (or similar).

Reply:
done in commit 3